### DISCUSSION TITLE
How to configure a hotkey for only the active screen

### DISCUSSION THREAD

**Discussion by chadacious** [debauchee/barrier#1383] [Q&A]
I have been trying this for some time now, thinking I must be missing something but can't figure it out. I started with the original synergy, then updated to barrier but the behavior is exactly the same.

My setup is a Windows 10 as server and MAC 11 as client.

I want to remap several keys for "only" when I am on the mac. For example, on windows Ctrl+Left moves one word left, but on Max it is Alt+Left. I can remap modifier keys to help with that, but then how could one map Ctrl+c to Meta+c on mac?

Using the Hotkeys seems like exactly what I needed and indeed, I set a bunch of them up and seemed great. But then I realized that anytime I used one of those keys on the server, it only ever executes that hotkey on the client. Yes, even though I am focused on the server screen. Checking the box for "only on these screens" and selecting the mac "still" only ever executes that key stroke on the mac, even when I'm focused on the server screen. Hard to believe this is a bug and must somehow be intended. But why would it be that checking the box for "only on these screens" not actually have any affect?

While on barrier, I have been sure to restart the server each time, although I didn't recall needing to do that for synergy. So I'm really at a loss. I tried Karabiner-elements, but sadly, it didn't seem to recognize the barrier keyboard so that didn't really work. Played with remapping keys using mac KeyBindings.dict, and that works a bit, but I don't really want to remap my mac keyboard unless I'm using barrier.

I was able to find that if I use my own configuration file, and remove the screens argument completely, it will now only active the keystroke on the "active" screen. But that doesn't solve the problem either since I still need the "normal" keystroke for the server. Tried setting up multiple hotkeys for the same key with something like this:
```
keystroke(Control+Left) = keystroke(Alt+Left) -- this now only sends the keystroke to the "active" screen
keystroke(Control+Left) = keystroke(Control+Left, IHMDEV08) -- I still want "normal" behavior on the windows server
```
But not surprisingly, it only respects the first one in the list.
It seems that what I need is to be able to create a "Hot Key" and exclude a screen. But I'm not seeing how to do that at present.

Thoughts?

**Comment by Rajveer86** [comment]
I had this exact issue, I also wanted to control a Mac from Windows, and make Mac behave in the same way. [#1883](https://github.com/debauchee/barrier/pull/1883) is a PR to make this work, @chadacious if you want I can give you a Windows build to test to see if this works for you. The PR gives a real-world example of how I use it.

**Reply by chadacious** [reply]
Great! Yes I'm interested. You could PM me from the contact info on my github profile.

**Reply by Rajveer86** [reply]
Awesome, I've created a release in my fork for you (and others) to [download](https://github.com/Rajveer86/barrier/releases/tag/PR-1883_test-build_1.0).
